appium/lib/screen-object/accessors.rb

```diff
@@ -462,82 +613,254 @@ def text_field(name,locator)
+      define_method("#{name}_cell_count") do
+        ScreenObject::AppElements::Table.new(locator).cell_count
+      end
+
+      # returns the underlying ScreenObject element to allow
+      # use of all inherited AppElements::Element methods
+      # this also makes it more consistent with PageObject gem
+      # which provides a _element method for any type of accessor
+      # @return [ScreenObject::AppElements::Table]
+      # table(:monthly_statement,"id~monthStatement")
+      # def get_monthly_statement_element
+      #   monthly_statement_element # This will not return the underlying ScreenObject::AppElements::Table object
+      #                  which can use all the inherited methods of ScreenObject::AppElements::Element
+      #                  like: .click, .value, .element etc. which are needed in certain cases
+      # end
+      define_method("#{name}_element") do
+        ScreenObject::AppElements::Table.new(locator)
+      end
+    end
+
+    # elements class generates all the methods related to general elements operation
+    def element(name, locator)
         #generates method for elements object
         define_method("#{name}") do
           ScreenObject::AppElements::Element.new(locator)
         end
-      end
 
+        define_method("#{name}?") do
+          ScreenObject::AppElements::Element.new(locator).exists?
+        end
+
+        # returns the underlying element collection to allow
+        # @return [Collection of element]
+        # element(:login,"xpath~//UITextField")
+        # def get_login_elements
+        #   login_elements # This will not return the underlying ScreenObject::AppElements::TextField object.
+        #                  the return elements can use all the inherited methods of the driver
+        #                  like: .click, .text etc. which are needed in certain cases
+        # end
+        define_method("#{name}_elements") do
+          ScreenObject::AppElements::Element.new(locator).elements
+        end
+
+        # generates method for scrolling down on both Android and iOS application screen and click on element.
+        # scroll to the first element with locator and click on it.
+        # this method will not return any value.
+        # element(:login_button,"UIButtonField/UIButtonFieldtext")
+        #   click_login_button
+
+        define_method("click_#{name}") do
+          # direction = options[:direction] || 'down'
+          ScreenObject::AppElements::Element.new(locator).scroll_element_to_view_click
+        end
+
+        # generates method for scrolling down on both Android and iOS application screen and click on element.
+        # scroll to the first element with locator and click on it.
+        # this method will not return any value.
+        # element(:login_button,"UIButtonField/UIButtonFieldtext")
+        #   scroll_down_to_click_login_button
+
+        define_method("scroll_down_to_click_#{name}") do
```

Comment:
You added this method `define_method("scroll_down_to_click_#{name}")` in classes: `Button`, `Text`, `Checkbox`, Element but not in `Image` & `TextField`

Reply:
I am not sure  the textfield would need the scroll down click. added to image.